context-propagation/src/main/java/io/micrometer/context/ContextAccessor.java

```diff
@@ -47,6 +47,16 @@
      */
     void readValues(READ sourceContext, Predicate<Object> keyPredicate, Map<Object, Object> readValues);
 
+    /**
+     * Read a single value from the source context.
+     * @param sourceContext the context to read from; the context type should be
+     * checked with {@link #canReadFrom(Class)} before this method is called
+     * @param key the key to use to look up the context value
+     * @return the value, if any
+     */
+    @Nullable
+    Object readValue(READ sourceContext, Object key);
```

Comment:
Maybe this could be <T> T readValue(...), wdyt?

Reply:
Yes, that would be useful.